src/Runner/Parallel/Process.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use React\ChildProcess\Process as ReactProcess;
+use React\EventLoop\LoopInterface;
+use React\EventLoop\TimerInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\WritableStreamInterface;
+
+/**
+ * Represents single process that is handled within parallel run.
+ * Inspired by:
+ *   - https://github.com/phpstan/phpstan-src/blob/9ce425bca5337039fb52c0acf96a20a2b8ace490/src/Parallel/Process.php
+ *   - https://github.com/phpstan/phpstan-src/blob/1477e752b4b5893f323b6d2c43591e68b3d85003/src/Process/ProcessHelper.php.
+ *
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class Process
+{
+    // Properties required for process instantiation
+    private string $command;
+    private LoopInterface $loop;
+    private int $timeoutSeconds;
+
+    // Properties required for process execution
+    private ?ReactProcess $process = null;
+    private ?WritableStreamInterface $in = null;
+
+    /** @var false|resource */
+    private $stdErr;
+
+    /** @var false|resource */
+    private $stdOut;
+
+    /** @var callable(mixed[]): void */
+    private $onData;
+
+    /** @var callable(\Throwable): void */
+    private $onError;
+
+    private ?TimerInterface $timer = null;
+
+    public function __construct(string $command, LoopInterface $loop, int $timeoutSeconds)
+    {
+        $this->command = $command;
+        $this->loop = $loop;
+        $this->timeoutSeconds = $timeoutSeconds;
+    }
+
+    /**
+     * @param callable(mixed[] $json): void                  $onData  callback to be called when data is received from the parallelisation operator
```

Comment:
do we have format of that data somewhere ?

Reply:
No, it can be any data actually. It depends on how you use `Clue\React\NDJson\Encoder::write()` in the worker.

Reply:
we don't send random data, right ?
I understand React library can accept anything, but we shall have our protocol and send data in explicit format. probably we already do, we can just document it ?

Reply:
It would be an union of different array shapes, it would be hard to grasp and easy to leave outdated, since there's no actual connection between this `@param` and places where we write data to encoder.

Right now you can easily find where specific kind of payload is written in `WorkerCommand`, and where it's consumed:

<img width="1667" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/7a148fde-52d4-4497-aed8-b524d7fb2c52">

The only approach I can think of is defining PHPStan aliases in this class, and use `_Payload1|_Payload2|_Payload3 $json` for better readability. But I don't see how to bind it with actual usage - I could import these types in `WorkerCommand`, but where to put these types? I would need to extract all the arrays, that are currently passed to `Encoder::write()`, to some methods that could have proper `@return _PayloadX`. I am not sure if it's worth it 🤔.

Do we really need it?